### PR TITLE
Read write functionality with hdf5

### PR DESCRIPTION
**Pull request by usmanwardag**
Adds read write functionality with hdf5 in both `io.py` and `lightcurve.py`.

### CODE REVIEW

File: stingray/lightcurve.py [hunk 1]
```diff
@@ -627,6 +637,19 @@ def read(self, filename, format_='pickle'):
...
 
-        self = io.read(filename, format_)
+        if format_ == 'ascii' or format_ == 'hdf5':
+            return io.read(filename, format_)
+
+        elif format_ == 'pickle':
+            self = io.read(filename, format_)
+
+        else:
+            logging.warn("Format not understood.")
```
Comment (OrkoHunter) [inline stingray/lightcurve.py]: For consistency, please use `utils.simon`. And add a new line.

File: stingray/io.py [hunk 2]
```diff
@@ -315,37 +325,74 @@ def _save_pickle_object(object, filename, **kwargs):
             pickle.dump(object, f)
 
 def _retrieve_pickle_object(filename):
+    """
+    Retrieves a pickled class object.
+
+    Parameters:
+    -----------
```
Comment (OrkoHunter) [inline stingray/io.py]: If I recall correctly, in doc strings, colon `:` is not required if the key word is underlined.

**Comment by usmanwardag** [comment]
@OrkoHunter Done. I think it can be merged now.

**Comment by OrkoHunter** [comment]
Awesome !

**Pull request merged**